src/reanimated2/Hooks.ts

```diff
@@ -163,17 +163,14 @@ function runAnimations(animation, timestamp, key, result, animationsActive) {
 function isAnimated(prop) {
   'worklet';
   if (Array.isArray(prop)) {
-    for (let i = 0; i < prop.length; ++i) {
-      const item = prop[i];
-      for (const key in item) {
-        if (item[key].onFrame !== undefined) {
-          return true;
-        }
-      }
-    }
-    return false;
+    return prop.some((item) => isAnimated(item));
+  } else if (prop?.onFrame !== undefined) {
+    return true;
+  } else if (typeof prop === 'object') {
+    return Object.values(prop).some((value) => isAnimated(value));
```

Comment:
There is a reason why for loops were used - they are just faster. And since we run that code on each frame - it's better to use the fastest code possible.